src/Docker/Manager/ContainerManager.php

```diff
@@ -172,6 +172,56 @@ public function start(Container $container, array $hostConfig = [])
         return $this;
     }
 
+
+    /**
+     * Copy files or folders from a container
+     *
+     * @param \Docker\Container $container
+     * @param string            $source file or folder name
+     *
+     * @throws \Docker\Exception\UnexpectedStatusCodeException
+     *
+     * @return Guzzle\Stream\Stream Tarfile stream
+     */
+    public function copy(Container $container, $source)
+    {
+        $response = $this->client->post(['/containers/{id}/copy', ['id' => $container->getId()]], [
+            'body'         => Json::encode(["Resource" => $source]),
+            'headers'      => ['content-type' => 'application/json'],
+        ]);
+
+        if ($response->getStatusCode() !== "200") {
+            throw UnexpectedStatusCodeException::fromResponse($response);
+        }
+
+        return $response->getBody();
+    }
+
+
+    /**
+     * Save files pulled by copy() to disk 
+     *
+     * @param \Docker\Container $container
+     * @param string            $source file or folder name
+     * @param string            $destination file
+     *
+     * @throws \Docker\Exception\UnexpectedStatusCodeException
+     *
+     * @return \Docker\Manager\ContainerManager
+     */
+
+    public function copyToDisk(Container $container, $source, $destination)
+    {
+        $stream = $this->copy($container, $source);
+
+        $output = fopen($destination, 'w+');
```

Comment:
We need to handle errors like 'permission denied' or 'file not found' (in case directory not exists). I suggest not to use fopen, it throws uncatchable warning in case of opening error.
Quick and dirty solution - mute fopen with `@` and then check $output is not `null`.
Better solution - open file by SPL `SplFileObject` in try-catch block.